flag.go

```diff
@@ -162,6 +162,12 @@ type Countable interface {
 	Count() int
 }
 
+// PersistentFlag is an interface to enable detection of flags which are persistent
+// through subcommands
+type PersistentFlag interface {
```

Comment:
This interface needs to be handled by the `SliceFlag` type.

```go
func (x *SliceFlag[T, S, E]) IsPersistent() bool {
    v, ok := x.Target.(PersistentFlag)
    return ok && v.IsPersistent()
}
```

^ or something

We shouldn't change the `SliceFlagTarget` interface for v2 (breaking change). For v3, ya might want to embed `PersistentFlag` into `SliceFlagTarget` for v3, along with removing the type assertion. CC @meatballhat

Reply:
Thank you @joeycumines!

Reply:
Updated. Thanks @joeycumines